xcat-inventory/xcclient/inventory/manager.py

```diff
@@ -99,100 +127,160 @@ def importObjs(self, objlist, obj_attr_dict,update=True):
         myclass = InventoryFactory.__InventoryClass__[self.objtype]
         myclass.loadschema(self.schemapath)
         dbdict = {}
+        objfiles={}
         for key, attrs in obj_attr_dict.items():
             if not objlist or key in objlist:
                 newobj = myclass.createfromfile(key, attrs)
+                objfiles[key]=newobj.getfilestosave()
                 dbdict.update(newobj.getdbdata())
         tabs=myclass.gettablist()
         if not update:
             self.getDBInst().cleartab(tabs)
         self.getDBInst().settab(dbdict)
+        return objfiles
 
 
+def dumpobj(objdict, fmt='json',location=None):
+    if not fmt or fmt.lower() == 'json':
+        dump2json(objdict,location)
+    else:
+        dump2yaml(objdict,location)
+
 def dump2yaml(xcatobj, location=None):
     if not location:
         print(yaml.dump(xcatobj, default_flow_style=False))
+    else:
+        f=open(location,'w')
+        print(yaml.dump(xcatobj, default_flow_style=False),file=f)
 
     #TODO: store in file or directory
 
 def dump2json(xcatobj, location=None):
     if not location:
         print(json.dumps(xcatobj, sort_keys=True, indent=4, separators=(',', ': ')))
+    else:
+        f=open(location,'w')
+        print(json.dumps(xcatobj, sort_keys=True, indent=4, separators=(',', ': ')),file=f)
 
     #TODO: store in file or directory
 
 def validate_args(args, action):
-    if args.type is not None and args.type.lower() not in InventoryFactory.getvalidobjtypes():
-        raise CommandException("Error: Invalid object type: \"%(t)s\"", t=args.type)
+    if args.type is not None:
+        objtypelist=[]
+        objtypelist.extend(args.type.split(','))
```

Comment:
You need to strip the input after split,  to considering "osimage,    node"

Reply:
updated